api/v1beta1/rabbitmqcluster_types.go

```diff
@@ -104,7 +104,8 @@ type RabbitmqClusterSpec struct {
 // Future secret backends could be Secrets Store CSI Driver.
 // If not configured, K8s Secrets will be used.
 type SecretBackend struct {
-	Vault *VaultSpec `json:"vault,omitempty"`
+	Vault          *VaultSpec `json:"vault,omitempty"`
+	ExternalSecret string     `json:"externalSecret,omitempty"`
```

Comment:
Looks like this later gets translated into a LocalObjectReference. Could this type just be a LocalObjectReference upfront? I'd be keen to find out if we get more upfront validation if we were to do that.

Reply:
By upfront validation, I mean would the Kubernetes API reject the request, rather than making it all the way to the controller before rejecting it.

Reply:
Hi @coro LocalObjectReference looks good! I modified the type!